nostr/relay.py

```diff
@@ -52,9 +52,9 @@ def __post_init__(self):
     def connect(self):
         self.ws.run_forever(
             sslopt=self.ssl_options,
-            http_proxy_host=self.proxy_config.host, 
-            http_proxy_port=self.proxy_config.port,
-            proxy_type=self.proxy_config.type
+            http_proxy_host=self.proxy_config.host if self.proxy_config is not None else None, 
```

Comment:
I think we can do a short-circuit technique to reduce verbosity.
```python
http_proxy_host=self.proxy_config.host or None
```
Applies to all three parameters

Reply:
`Relay` is instantiated by `RelayManager` which by default will pass `proxy_config=None`. So this check in `connect()` can't assume there's an object there. That's the bug this PR is trying to fix. The default value on `Relay` (before this PR) tries to set an empty `RelayProxyConnectionConfig` but it's bypassed because `RelayManager` IS explicitly passing in a value, but it just happens to be `None`.

Two alternatives:
Option 1: Initialize `self.proxy_config` to an empty `RelayProxyConnectionConfig` in the `__post_init__()` if the `Relay` is instantiated with `None`. But as I mentioned elsewhere, I don't like this; If there's no proxy, there should be no `proxy_config` object.

Option 2: `RelayManager`'s default could be changed to an empty `RelayProxyConnectionConfig`. I dislike this even more for similar reasons.

Reply:
Okay, I understand. We will stick with what you have here. Thanks